apiml-security-common/src/main/java/org/zowe/apiml/security/common/login/LoginRequest.java

```diff
@@ -10,50 +10,59 @@
 
 package org.zowe.apiml.security.common.login;
 
+import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 import org.springframework.security.core.Authentication;
 
+import java.util.Arrays;
+
+import static org.zowe.apiml.security.SecurityUtils.readPassword;
+
 /**
  * Represents the login JSON with credentials
  */
 @Data
 @NoArgsConstructor
+@AllArgsConstructor
 public class LoginRequest {
-    private String username;
-    private String password;
-    private String newPassword;
 
-    public LoginRequest(String username, String password) {
-        this.username = username;
-        this.password = password;
-    }
+    private String username;
+    private char[] password;
+    private char[] newPassword;
 
-    public LoginRequest(String username, String password, String newPassword) {
+    public LoginRequest(String username, char[] password) {
         this.username = username;
         this.password = password;
-        this.newPassword = newPassword;
     }
 
-    public static String getPassword(Authentication authentication) {
-        String password;
+    public static char[] getPassword(Authentication authentication) {
         if (authentication.getCredentials() instanceof LoginRequest) {
             LoginRequest loginRequest = (LoginRequest) authentication.getCredentials();
-            password = loginRequest.getPassword();
-        } else {
-            password = (String) authentication.getCredentials();
+            return readPassword(loginRequest.getPassword());
         }
-        return password;
+
+        return readPassword(authentication.getCredentials());
     }
 
-    public static String getNewPassword(Authentication authentication) {
-        String password;
+    public static char[] getNewPassword(Authentication authentication) {
         if (authentication.getCredentials() instanceof LoginRequest) {
             LoginRequest loginRequest = (LoginRequest) authentication.getCredentials();
-            password = loginRequest.getNewPassword();
-        } else {
-            password = null;
+            return readPassword(loginRequest.getNewPassword());
         }
-        return password;
+        return null;
```

Comment:
What about 
```suggestion
        return new char[0];
```

Reply:
I'd say this changes how it used to work since it was already returning null, but if it doesn't break I'm not against it